branchLayoutImpl/src/main/java/com/virtuslab/branchlayout/impl/BranchLayoutFileParser.java

```diff
@@ -19,20 +21,37 @@
 @RequiredArgsConstructor
 @Getter
 public class BranchLayoutFileParser implements IBranchLayoutParser {
+  private static final LambdaLogger LOG = LambdaLoggerFactory.getLogger("branchLayout");
+
   private final Path path;
   private Character indentCharacter = ' ';
   @NonNegative
   private int levelWidth = 0;
 
   public BranchLayout parse() throws BranchLayoutException {
+    LOG.debug("Enter BranchLayout#parse");
+    LOG.debug(() -> "Branch layout file path: ${path}");
+
     List<BaseBranchLayoutEntry> roots = List.empty();
-    List<String> lines = getFileLines().reject(String::isBlank);
+    List<String> lines = getFileLines();
+    List<String> linesWithoutBlank = lines.reject(String::isBlank);
+
+    LOG.debug(() -> "${lines.length()} line(s) fund");
 
     deriveIndentCharacter(lines);
 
-    if (!lines.isEmpty()) {
+    LOG.debug(() -> "Indent character is ${indentCharacter.equals('\\t') ? \"TAB\" : indentCharacter.equals(' ') " +
+        "? \"SPACE\" : \"\\\"\" + indentCharacter + \"\\\"\"}");
```

Comment:
Maybe change the quote character being logged from `"` to `'` instead to reduce the _escapism_